app/test/vesting_test.go

```diff
@@ -0,0 +1,632 @@
+}
+
+// initDelayedVestingAccounts initializes delayed vesting accounts for the VestingModuleTestSuite.
+// It generates the specified number of account names and stores them in the accounts map with
+// DelayedVestingAccountType as the key.
+func (s *VestingModuleTestSuite) initDelayedVestingAccounts(count int) testnode.GenesisOption {
+	initCoinsForGasFee := sdk.NewCoin(app.BondDenom, sdk.NewInt(initBalanceForGasFee))
+
+	names := testfactory.GenerateAccounts(count)
+	s.accounts.Store(DelayedVestingAccountType, accountDispenser{names: names})
+	vAccounts, balances := testfactory.GenerateDelayedVestingAccounts(s.kr, names, initCoinsForGasFee)
+	gAccounts := authtypes.GenesisAccounts{}
+	for i := range vAccounts {
+		gAccounts = append(gAccounts, authtypes.GenesisAccount(vAccounts[i]))
+	}
+
+	return func(gs map[string]json.RawMessage) map[string]json.RawMessage {
+		gs, err := AddGenesisAccountsWithBalancesToGenesisState(gs, gAccounts, balances)
+		assert.NoError(s.T(), err)
+		return gs
+	}
+}
+
+// initPeriodicVestingAccounts function initializes periodic vesting accounts for testing purposes.
+// It takes the count of accounts as input and returns a GenesisOption. It generates account names,
+// base accounts, and balances. It defines vesting periods and creates vesting accounts based on the
+// generated data. The function then updates the genesis state with the generated vesting accounts
+// and balances. The startTime of each account increases progressively to ensure some accounts have
+// locked balances, catering to the testing requirements.
+func (s *VestingModuleTestSuite) initPeriodicVestingAccounts(count int) testnode.GenesisOption {
+	initCoinsForGasFee := sdk.NewCoin(app.BondDenom, sdk.NewInt(initBalanceForGasFee))
+
+	names := testfactory.GenerateAccounts(count)
+	s.accounts.Store(PeriodicVestingAccountType, accountDispenser{names: names})
+	vAccounts, balances := testfactory.GeneratePeriodicVestingAccounts(s.kr, names, initCoinsForGasFee)
+	gAccounts := authtypes.GenesisAccounts{}
+	for i := range vAccounts {
+		gAccounts = append(gAccounts, authtypes.GenesisAccount(vAccounts[i]))
+	}
+
+	return func(gs map[string]json.RawMessage) map[string]json.RawMessage {
+		gs, err := AddGenesisAccountsWithBalancesToGenesisState(gs, gAccounts, balances)
+		assert.NoError(s.T(), err)
+		return gs
+	}
+}
+
+// initContinuousVestingAccounts function initializes continuous vesting accounts for testing purposes.
+// It takes the count of accounts as input and returns a GenesisOption. It generates account names,
+// base accounts, and balances. It defines start & end times to creates vesting accounts. The function
+// then updates the genesis state with the generated vesting accounts and balances. The start & endTime
+// of each account increases progressively to ensure some accounts have locked balances, catering to the
+// testing requirements.
+func (s *VestingModuleTestSuite) initContinuousVestingAccounts(count int) testnode.GenesisOption {
```

Comment:
[optional][refactor]
I guess this can be refactored to: 
```
func (s *VestingModuleTestSuite) initVestingAccounts(accountType accountType, count int) testnode.GenesisOption {
    ...
    switch accountType {
	case RegularAccountType:
                ....
	case DelayedVestingAccountType:
		....
	}
    ....
```

So that we don't have duplicate functions for every vesting account type